app/main.py

```diff
@@ -1,2 +1,94 @@
-class Dictionary:
-    pass
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Hashable
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+
+
+class Dictionary(object):
+
+    def __init__(self, capacity: int = 8) -> None:
+        self._capacity = capacity
+        self._size = 0
+        self._table: list[None | Node] = [None] * self._capacity
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self.get_index(key)
+
+        if not self._table[index]:
+            raise KeyError(f"Key: {key} is not found!")
+
+        return self._table[index].value
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.get_index(key)
+        if not self._table[index]:
+            if self._size + 1 >= self._capacity * 2 / 3:
```

Comment:
It is better to define load factor in the `__init__` to be able to change it from there, not from the code itself